open-api/rest-catalog-open-api.yaml

```diff
@@ -532,6 +532,103 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/namespaces/{namespace}/tables/{table}/preplan:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    post:
+      tags:
+        - Catalog API
+      summary: Intiate a preplanning process on the table to find plan-tasks relevant to a specific query.
+      description:
+        When a user submits a query, we can do a "preplan" to find the relevant plan-tasks
+        based on the user's selected columns, and filters. The plan-tasks can then be used during "plan"
+        for distributed planning for performance gain.
+      operationId: prePlan
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PrePlanTableRequest'
```

Comment:
just nit about the word "preplan", looks like it can exist either as one whole word, or two words `pre-plan`: https://www.merriam-webster.com/dictionary/preplan

I think we should be consistent for this wording in the spec. If `preplan` is the word we will always use, We should always stick to `Preplan` instead of `PrePlan` for camel casing.

Reply:
Ack will fix this.